astropy/io/fits/column.py

```diff
@@ -1429,6 +1429,12 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
+            # On Python 2, force the `name` to `str`
+            # because Numpy structured arrays can't handle unicode `name`
+            # See https://github.com/astropy/astropy/issues/5204
+            if PY2:  # pragma: py2
+                name = str(name)
```

Comment:
@saimn - Thanks for the review and checking!

I'm not familiar with unicode / str much. I only put `name = str(name)` because that was mentioned as a potential fix int the corresponding Numpy issue discussion and checked that it works for me for that one example in all cases:
- Python 2 with `unicode_literals`
- Python 2 without `unicode_literals`
- Python 3